src/Services/FHIR/DocumentReference/FhirPatientDocumentReferenceService.php

```diff
@@ -0,0 +1,211 @@
+use OpenEMR\FHIR\R4\FHIRElement\FHIRIdentifier;
+use OpenEMR\FHIR\R4\FHIRElement\FHIRMeta;
+use OpenEMR\FHIR\R4\FHIRElement\FHIRPeriod;
+use OpenEMR\FHIR\R4\FHIRElement\FHIRString;
+use OpenEMR\FHIR\R4\FHIRElement\FHIRUrl;
+use OpenEMR\FHIR\R4\FHIRResource\FHIRDocumentReference\FHIRDocumentReferenceContent;
+use OpenEMR\FHIR\R4\FHIRResource\FHIRDocumentReference\FHIRDocumentReferenceContext;
+use OpenEMR\Services\DocumentService;
+use OpenEMR\Services\FHIR\FhirCodeSystemUris;
+use OpenEMR\Services\FHIR\FhirOrganizationService;
+use OpenEMR\Services\FHIR\FhirProvenanceService;
+use OpenEMR\Services\FHIR\FhirServiceBase;
+use OpenEMR\Services\FHIR\Traits\FhirServiceBaseEmptyTrait;
+use OpenEMR\Services\FHIR\Traits\PatientSearchTrait;
+use OpenEMR\Services\FHIR\UtilsService;
+use OpenEMR\Services\Search\FhirSearchParameterDefinition;
+use OpenEMR\Services\Search\SearchFieldType;
+use OpenEMR\Services\Search\ServiceField;
+use OpenEMR\Validators\ProcessingResult;
+
+class FhirPatientDocumentReferenceService extends FhirServiceBase
+{
+    use FhirServiceBaseEmptyTrait;
+    use PatientSearchTrait;
+
+    /**
+     * @var DocumentService
+     */
+    private $service;
+
+    public function __construct($fhirApiURL = null)
+    {
+        parent::__construct($fhirApiURL);
+        $this->service = new DocumentService();
+    }
+
+
+    public function supportsCategory($category)
+    {
+        // we have no category definitions right now for patient
+        return false;
+    }
+
+
+    public function supportsCode($code)
+    {
+        // we don't support searching by code
+        return false;
+    }
+
+    protected function loadSearchParameters()
+    {
+        return  [
+            'patient' => $this->getPatientContextSearchField(),
+            'date' => new FhirSearchParameterDefinition('date', SearchFieldType::DATETIME, ['date']),
+            '_id' => new FhirSearchParameterDefinition('_id', SearchFieldType::TOKEN, [new ServiceField('drive_uuid', ServiceField::TYPE_UUID)]),
+        ];
+    }
+
+    protected function searchForOpenEMRRecords($openEMRSearchParameters): ProcessingResult
+    {
+        if (isset($openEMRSearchParameters['category'])) {
+            // we have nothing with this category so we are going to return nothing as we never should have gotten here
+
+            unset($openEMRSearchParameters['category']);
+        }
+        return $this->service->search($openEMRSearchParameters);
+    }
+
+    public function parseOpenEMRRecord($dataRecord = array(), $encode = false)
+    {
+        $docReference = new FHIRDocumentReference();
+        $meta = new FHIRMeta();
+        $meta->setVersionId('1');
+        $meta->setLastUpdated(gmdate('c'));
+        $docReference->setMeta($meta);
+
+        $id = new FHIRId();
+        // TODO: @adunsulag after playing around with this we definitely need a document uuid, drive_uuid doesn't work
+        // as it only applies to anything not couchdb...
```

Comment:
agreed, as i recall the drive_uuid is just meant to be the name of the actual filename stored on the drive or couchdb. would not use it as the official uuid identifier.

Reply:
should be resolved now.